### DISCUSSION TITLE
Install ray library causes app to result in html error 500 php 8.2

### DISCUSSION THREAD

**Discussion by colmanbyrne** [spatie/ray#788] [Q&A]
When I install the ray library into my laravel  app with composer require spatie/laravel-ray
My app starts failing with 500 error , when I remove it without changing app , pages work again 

Laravel v10.4.1 (PHP v8.2.2)

any ideas what's causing this ?

[2023-03-28 19:01:51] local.ERROR: Uncaught Error: Call to undefined function Spatie\Ray\curl_init() in D:\Projects\newiwaionline365\vendor\spatie\ray\src\Client.php:148
Stack trace:
#0 D:\Projects\newiwaionline365\vendor\spatie\ray\src\Client.php(143): Spatie\Ray\Client->getCurlHandle('get', 'http://localhos...')
#1 D:\Projects\newiwaionline365\vendor\spatie\ray\src\Client.php(73): Spatie\Ray\Client->getCurlHandleForUrl('get', '')
#2 D:\Projects\newiwaionline365\vendor\spatie\ray\src\Ray.php(793): Spatie\Ray\Client->send(Object(Spatie\Ray\Request))
#3 D:\Projects\newiwaionline365\vendor\spatie\laravel-ray\src\Ray.php(507): Spatie\Ray\Ray->sendRequest(Array, Array)
#4 D:\Projects\newiwaionline365\vendor\spatie\laravel-ray\src\Watchers\ApplicationLogWatcher.php(33): Spatie\LaravelRay\Ray->sendRequest(Object(Spatie\Ray\Payloads\ApplicationLogPayload))
#5 D:\Projects\newiwaionline365\vendor\laravel\framework\src\Illuminate\Events\Dispatcher.php(421): Spatie\LaravelRay\Watchers\ApplicationLogWatcher->Spatie\LaravelRay\Watchers\{closure}(Object(Illuminate\Log\Events\MessageLogged))
#6 D:\Projects\newiwaionline365\vendor\laravel\framework\src\Illuminate\Events\Dispatcher.php(249): Illuminate\Events\Dispatcher->Illuminate\Events\{closure}('Illuminate\\Log\\...', Array)
#7 D:\Projects\newiwaionline365\vendor\laravel\framework\src\Illuminate\Log\Logger.php(248): Illuminate\Events\Dispatcher->dispatch('Illuminate\\Log\\...')
#8 D:\Projects\newiwaionline365\vendor\laravel\framework\src\Illuminate\Log\Logger.php(189): Illuminate\Log\Logger->fireLogEvent('error', 'Call to undefin...', Array)
#9 D:\Projects\newiwaionline365\vendor\laravel\framework\src\Illuminate\Log\Logger.php(97): Illuminate\Log\Logger->writeLog('error', 'Call to undefin...', Array)
#10 D:\Projects\newiwaionline365\vendor\laravel\framework\src\Illuminate\Log\LogManager.php(667): Illuminate\Log\Logger->error('Call to undefin...', Array)
#11 D:\Projects\newiwaionline365\vendor\laravel\framework\src\Illuminate\Foundation\Exceptions\Handler.php(274): Illuminate\Log\LogManager->error('Call to undefin...', Array)
#12 D:\Projects\newiwaionline365\vendor\laravel\framework\src\Illuminate\Foundation\Bootstrap\HandleExceptions.php(182): Illuminate\Foundation\Exceptions\Handler->report(Object(Error))
#13 D:\Projects\newiwaionline365\vendor\laravel\framework\src\Illuminate\Foundation\Bootstrap\HandleExceptions.php(250): Illuminate\Foundation\Bootstrap\HandleExceptions->handleException(Object(Error))
#14 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->Illuminate\Foundation\Bootstrap\{closure}(Object(Error))
#15 {main}
  thrown {"userId":3,"exception":"[object] (Symfony\\Component\\ErrorHandler\\Error\\FatalError(code: 0): Uncaught Error: Call to undefined function Spatie\\Ray\\curl_init() in D:\\Projects\\newiwaionline365\\vendor\\spatie\\ray\\src\\Client.php:148
Stack trace:
#0 D:\\Projects\\newiwaionline365\\vendor\\spatie\\ray\\src\\Client.php(143): Spatie\\Ray\\Client->getCurlHandle('get', 'http://localhos...')
#1 D:\\Projects\\newiwaionline365\\vendor\\spatie\\ray\\src\\Client.php(73): Spatie\\Ray\\Client->getCurlHandleForUrl('get', '')
#2 D:\\Projects\\newiwaionline365\\vendor\\spatie\\ray\\src\\Ray.php(793): Spatie\\Ray\\Client->send(Object(Spatie\\Ray\\Request))
#3 D:\\Projects\\newiwaionline365\\vendor\\spatie\\laravel-ray\\src\\Ray.php(507): Spatie\\Ray\\Ray->sendRequest(Array, Array)
#4 D:\\Projects\\newiwaionline365\\vendor\\spatie\\laravel-ray\\src\\Watchers\\ApplicationLogWatcher.php(33): Spatie\\LaravelRay\\Ray->sendRequest(Object(Spatie\\Ray\\Payloads\\ApplicationLogPayload))
#5 D:\\Projects\\newiwaionline365\\vendor\\laravel\\framework\\src\\Illuminate\\Events\\Dispatcher.php(421): Spatie\\LaravelRay\\Watchers\\ApplicationLogWatcher->Spatie\\LaravelRay\\Watchers\\{closure}(Object(Illuminate\\Log\\Events\\MessageLogged))
#6 D:\\Projects\\newiwaionline365\\vendor\\laravel\\framework\\src\\Illuminate\\Events\\Dispatcher.php(249): Illuminate\\Events\\Dispatcher->Illuminate\\Events\\{closure}('Illuminate\\\\Log\\\\...', Array)
#7 D:\\Projects\\newiwaionline365\\vendor\\laravel\\framework\\src\\Illuminate\\Log\\Logger.php(248): Illuminate\\Events\\Dispatcher->dispatch('Illuminate\\\\Log\\\\...')
#8 D:\\Projects\\newiwaionline365\\vendor\\laravel\\framework\\src\\Illuminate\\Log\\Logger.php(189): Illuminate\\Log\\Logger->fireLogEvent('error', 'Call to undefin...', Array)
#9 D:\\Projects\\newiwaionline365\\vendor\\laravel\\framework\\src\\Illuminate\\Log\\Logger.php(97): Illuminate\\Log\\Logger->writeLog('error', 'Call to undefin...', Array)
#10 D:\\Projects\\newiwaionline365\\vendor\\laravel\\framework\\src\\Illuminate\\Log\\LogManager.php(667): Illuminate\\Log\\Logger->error('Call to undefin...', Array)
#11 D:\\Projects\\newiwaionline365\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Exceptions\\Handler.php(274): Illuminate\\Log\\LogManager->error('Call to undefin...', Array)
#12 D:\\Projects\\newiwaionline365\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Bootstrap\\HandleExceptions.php(182): Illuminate\\Foundation\\Exceptions\\Handler->report(Object(Error))
#13 D:\\Projects\\newiwaionline365\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Bootstrap\\HandleExceptions.php(250): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleException(Object(Error))
#14 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}(Object(Error))
#15 {main}
  thrown at D:\\Projects\\newiwaionline365\\vendor\\spatie\\ray\\src\\Client.php:148)
[stacktrace]
#0 {main}
"}

**Comment by freekmurze** [comment]
The Ray package relies on curl, which is available on the vast majority of PHP installations. The error message indicates that curl or PHP's curl extensions is not available on your system.

**Comment by colmanbyrne** [comment]
Hi

Ok that I can fix


Thanks


Colman
On Wed 29 Mar 2023 at 07:57, Freek Van der Herten ***@***.***>
wrote:

> The Ray package relies on curl, which is available on the vast majority of
> PHP installations. The error message indicates that curl or PHP's curl
> extensions is not available on your system.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/spatie/ray/discussions/788#discussioncomment-5462037>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAYA4W5YWUSD5AQKQJF3N3DW6PMMVANCNFSM6AAAAAAWK7MI6U>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
Regards


Colman Byrne

**Reply by colmanbyrne** [reply]
I just checked and curl is installed , however when I go back to PKP 8.1.0 all works ok , with PHP 8.2.2 appears to have a problem

**Reply by freekmurze** [reply]
I think it's still caused by something on your side, as our [testsuite is passing](https://github.com/spatie/ray/actions/runs/4340992239) and I didn't get any similar reports by others.

Should you find something wrong with our package, feel free to submit a PR to fix it.
